saltnanny/salt_nanny.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+import logging
+import math
+import time
+import os
+
+from salt_return_parser import SaltReturnParser
+from salt_nanny_client import SaltNannyClientFactory
+
+
+class SaltNanny:
+
+    log = logging.getLogger('saltnanny')
+    log_fmt = '%(asctime)s %(levelname)s %(message)s'
+
+    def __init__(self, cache_config, target_log_file=None, fun='state.highstate'):
+        self.fun = fun
+        self.initial_jids = {}
+        self.completed_minions = {}
+        self.wait_time = 15
+        self.setup_logging(target_log_file)
+        self.cache_client = SaltNannyClientFactory.factory(
+            cache_config['type'], cache_config['host'], cache_config['port'], cache_config['db'])
+        self.parser = SaltReturnParser(self.cache_client)
+
+    def initialize(self, minion_list):
+        if minion_list:
+            self.minion_list = minion_list
+            for minion in minion_list:
+                self.initial_jids[minion] = self.cache_client.get_latest_jid(minion, self.fun)
+        else:
+            raise ValueError("No Minions provided. SaltNanny has no minions to babysit.")
+
+    def track_returns(self, max_attempts=17):
+        pending_minion_list = list(self.minion_list)
+
+        for i in xrange(0, max_attempts):
+            self.log.info('SaltNanny is checking for minion returns in External Job Cache - Attempt: {0}'.format(i))
+            pending_minion_list = [minion for minion in pending_minion_list if minion not in self.completed_minions.keys()]
+
+            if pending_minion_list:
+                for minion in pending_minion_list:
+                    latest_jid = self.cache_client.get_latest_jid(minion, self.fun)
+                    if latest_jid == self.initial_jids[minion]:
+                        self.log.info('SaltNanny did not find a new JID for minion :{0}'.format(minion))
+                        break
+                    else:
+                        self.log.info('SaltNanny found a New JID for minion:{0} JID:{1}'.format(minion, latest_jid))
+                        self.completed_minions[minion] = latest_jid
+                time.sleep(self.get_wait_time(i))
```

Comment:
would it be helpful to log that we are sleeping... i.e "Sleeping for x...". In this way, there would be no guessing as to what happening to the process